packages/block-library/src/index.js

```diff
@@ -229,6 +228,23 @@ const getAllBlocks = () => {
 		queryTitle,
 		postAuthorBiography,
 	];
+
+	// Only add the classic block in WP Context
+	// and when either of the following is true:
+	// - TinyMCE experiment is disabled;
+	// - there is a classic block in the current post's content;
+	// - the query argument to require TinyMCE is set.
```

Comment:
might consider rephrasing this; I found tracking some of the negatives difficult, and "either" normally implies one of two

```js
// When in a WordPress context, conditionally
// add the classic block and TinyMCE editor
// under any of the following conditions:
//   - the current post contains a classic block
//   - TinyMCE hasn't been intentionally disabled
//   - a query argument specifies that TinyMCE should be loaded
```

on this point I guess I have two follow-up questions: would these conditions be better as named variables so that the comment doesn't need to encode the logic? should we be loading TinyMCE if `__experimentalDisableTinymce` is set?

```js
const runningInWpContext = window?.wp?.oldEditor;

if ( ! runningInWpContext ) {
	return blocks.filter( Boolean );
}

const postContainsClassicBlock = window?.wp?.needsClassicBlock;
const tinymceNormallyAllowed = ! window?.__experimentalDisableTinymce;
const queryRequiresTinymce = !! new URLSearchParams( window?.location?.search ).get( 'requiresTinymce' );

if ( postContainsClassicBlock || tinymceNormallyAllowed || queryRequiresTinymce ) {
	blocks.push( classic );
}

return blocks.filter( Boolean );
```

Reply:
I really appreciate you taking the time to improve the comment 🙌 

In https://github.com/WordPress/gutenberg/pull/50387/commits/1c5199025caab2a9e270be6f92de0ffaa6019f79 I've reused your suggestion and just slightly updated it to clarify that the **experiment** needs to be disabled as one of the conditions, not TinyMCE. 

Then on your suggestion to change the comment to using variables, I think this makes it less readable, and we have enough specific information to warrant having a comment. I think it would be more difficult to convey all that information within variable names. I think the new version of the comment is doing a great job with that 👍

Reply:
thanks!

I guess I still find this experiment condition confusing, which is why I had inverted the variable to avoid having to follow the negations. this is just a personal response though; not a demand. each time I read it, I do spend what seems like too long trying to understand exactly what it's trying to check

```
TinyMCE experiment hasn't been intentionally disabled
```

so this is an experiment to use TinyMCE?
if it's disabled, that means don't load the TinyMCE editor, right?

```
! window?.__experimentalDisableTinymce
```

Oh no, wait, if the experiment to _disable_ TinyMCE is _not_ activated…

🤔 

What about this? It's closer to the code check anyhow. Maybe it's just the mention of "experiment" without indicating that the experiment is to disable the editor that leads to confusion.

> Experiment to disable TinyMCE isn't active.

Reply:
Nice, @dmsnell, agreed that what you suggested is much better phrasing. 

Updated in 7b3ddaca89f4497581258d5fae66fe46a98c7b41.

Thank you 🙌